content/docs/details/scaffolder/index.md

```diff
@@ -0,0 +1,204 @@
+---
+title: Scaffolder
+publishedDate: '2022-05-13T10:53:00.0Z'
+description: Initialize projects and trigger automation from within Backstage. You can prompt parameter inputs and combine scaffolder actions in oder to build a scaffolder template to run within Backstage.
+---
+
+# Overview
+Imagine you’re an engineer looking to create a new microservice. You want to get started as quickly as possible, with minimal boilerplate and red-tape to jump through. At the same time, engineering organizations benefit from having consistency in production, and often put gates in place to enforce it.
+
+Instead of creating blockers for engineering teams, Spotify use the scaffolder and quickly create new microservices, while helping to ensure that production remains mostly consistent.
+
+Engineers can choose a pre-defined software template, fill out a few form fields to provide values like the name of the GitHub repo that the new service will occupy, and click a button to run the template and create a new service.
+
+# Components of a Template
+You can configure a Scaffolder _template_ to be triggered on demand by Backstage users. Your template contains one or more _steps_ which run sequentially during execution of a template. You may optionally specify All _steps_ during the template execution are running inside an ephemeral container that is destroyed after the execution completes.
+
+A Scaffolder _template_ is a configurable process that will run one or more Scaffolder _steps_. The template will be run when a user visits the "Create Component" page in Backstage. `https://<tenant-name>.roadie.so/create`.
+
+![create-a-new-component](./create-a-new-component.png)
+
+Templates are defined by a Backstage Entity YAML file with a _Template_ kind and imported into the Backstage catalog.  You can create multiple templates, each of which can perform a different set of steps. For example, you can have one _template_ that creates a React application, and another that creates a serverless app.
+
+Here is an example of a very basic Scaffolder _template_ that prompts the user for a name, and then prints back the text "Hello, name!"
+
+```yaml
+apiVersion: scaffolder.backstage.io/v1beta3
+kind: Template
+metadata:
+  name: hello-world-template
+  title: Hello World
+  description: Says Hello to a specified name.
+spec:
+  owner: backstage/techdocs-core
+  type: service
+
+  parameters:
+    - title: You are about to say hello to your first Backstage Template
+      required:
+        - name
+      properties:
+        name:
+          title: Name
+          type: string
+          description: Name to say hello to
+          ui:autofocus: true
+          ui:options:
+            rows: 5
+
+  steps:
+    - id: log-message
+      name: Log Message
+      action: debug:log
+      input:
+        message: 'Hello, {{ parameters.name }}!'
```

Comment:
```suggestion
        message: 'Hello, ${{ parameters.name }}!'
```

According to https://backstage.io/docs/features/software-templates/migrating-from-v1beta2-to-v1beta3